hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -462,6 +471,21 @@ public <T> T dispatchChildTransaction(
         return doDispatchChildTransaction(syntheticPayer, txBody, childRecordBuilder, recordBuilderClass, callback);
     }
 
+    @Override
+    public @NonNull Fees dispatchComputeFees(@NonNull final TransactionBody txBody, @NonNull final AccountID payerId) {
+        var bodyToDispatch = txBody;
+        if (!txBody.hasTransactionID()) {
+            // Legacy mono fee calculators frequently estimate an entity's lifetime using the epoch second of the
+            // transaction id/ valid start as the current consensus time; ensure those will behave sensibly here
+            bodyToDispatch = txBody.copyBuilder()
+                    .transactionID(TransactionID.newBuilder()
+                            .transactionValidStart(Timestamp.newBuilder()
+                                    .seconds(consensusNow().getEpochSecond())))
```

Comment:
Down at `AbstractContractXTest` L137 `nanos()` is set as well as `seconds()` - intentional difference?

Reply:
Entity lifetimes are only computed relative to second, but to reduce possible confusion, added `nanos()` here.